core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Provides an executor interface which is used for potentially long-running tasks which may block I/O threads.
+ */
+public interface BlockingTaskExecutor extends ScheduledExecutorService {
+
+    /**
+     * Returns a {@link BlockingTaskExecutor} with a 60s timeout and unbounded work queue.
+     */
+    static BlockingTaskExecutor of() {
+        return builder().build();
```

Comment:
Should we return a new executor every time or a singleton? I guess singleton might be better? i.e. return the same instance with `CommonPools.blockingTaskExecutor()`

Reply:
Yeah, in Armeria, `of` methods return consistent singleton object. I guess `CommonPools.blockingTaskExecutor()` also good idea.

Reply:
One simple qq: https://github.com/line/armeria/pull/3389/commits/45829486d133ad5177ac8738e7ad17ca5c332e31#diff-36e3bf996d3f5f5a9e4aa7c564a70f9e6241e70af60d28f67e093249e0260cbeR32

Can we guarantee this object is `BlockingTaskExecutor` type right ? I always feel uncomfortable using type casting in such place :)

Reply:
Yes, I think so. We can change the return type to `BlockingTaskExecutor` when Armeria releases 2.0. By the way, could leave a comment why `CommonPools.blockingTaskExecutor()` returns `ScheduledExecutorService` so that we don't forget the breaking change in Armeria 2.0.